aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -649,24 +649,7 @@ Tensor stft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop
   // default_init hop_length and win_length
   auto hop_length = hop_lengthOpt.value_or(n_fft >> 2);
   auto win_length = win_lengthOpt.value_or(n_fft);
-  const bool return_complex = return_complexOpt.value_or(
-      self.is_complex() || (window.defined() && window.is_complex()));
-  if (!return_complex) {
-    if (!return_complexOpt.has_value()) {
-      TORCH_WARN_ONCE(
-        "stft will soon require the return_complex parameter be given for real inputs, "
-        "and will further require that return_complex=True in a future PyTorch release."
-      );
-    }
-
 
-    // TORCH_WARN_ONCE(
-    //     "stft with return_complex=False is deprecated. In a future pytorch "
-    //     "release, stft will return complex tensors for all inputs, and "
-    //     "return_complex=False will raise an error.\n"
-    //     "Note: you can still call torch.view_as_real on the complex output to "
-    //     "recover the old return format.");
-  }
```

Comment:
The next step in the deprecation process would be this:

```cpp
  const bool return_complex = return_complexOpt.value_or(
      self.is_complex() || (window.defined() && window.is_complex()));
  if (!return_complex) {
    TORCH_CHECK(
        return_complexOpt.has_value(),
        "stft will requires the return_complex parameter be given for real inputs");


    TORCH_WARN_ONCE(
        "stft with return_complex=False is deprecated. In a future pytorch "
        "release, stft will return complex tensors for all inputs, and "
        "return_complex=False will raise an error.\n"
        "Note: you can still call torch.view_as_real on the complex output to "
        "recover the old return format.");
  }

```

Or if we're okay with breaking BC without warning then `TORCH_CHECK(return_complex, ...)`.

Reply:
That seems like a different PR. This one was meant to do something different. I will start over.